lib/omniauth/strategies/azure_activedirectory_v2.rb

```diff
@@ -14,41 +16,45 @@ class AzureActivedirectoryV2 < OmniAuth::Strategies::OAuth2
       args [:tenant_provider]
 
       def client
-        if options.tenant_provider
-          provider = options.tenant_provider.new(self)
-        else
-          provider = options  # if pass has to config, get mapped right on to options
-        end
+        provider = if options.tenant_provider
+                     options.tenant_provider.new(self)
+                   else
+                     options # if pass has to config, get mapped right on to options
+                   end
 
         options.client_id = provider.client_id
         options.client_secret = provider.client_secret
         options.tenant_id =
-            provider.respond_to?(:tenant_id) ? provider.tenant_id : 'common'
+          provider.respond_to?(:tenant_id) ? provider.tenant_id : 'common'
         options.base_azure_url =
-            provider.respond_to?(:base_azure_url) ? provider.base_azure_url : BASE_AZURE_URL
+          provider.respond_to?(:base_azure_url) ? provider.base_azure_url : BASE_AZURE_URL
 
         options.authorize_params = provider.authorize_params if provider.respond_to?(:authorize_params)
-        options.authorize_params.domain_hint = provider.domain_hint if provider.respond_to?(:domain_hint) && provider.domain_hint
-        options.authorize_params.prompt = request.params['prompt'] if defined? request && request.params['prompt']
-        options.authorize_params.scope = (provider.scope if provider.respond_to?(:scope) && provider.scope) || DEFAULT_SCOPE
+        if provider.respond_to?(:domain_hint) && provider.domain_hint
+          options.authorize_params.domain_hint = provider.domain_hint
+        end
+        options.authorize_params.prompt = request.params['prompt'] if defined?(request) && request.params['prompt']
+        options.authorize_params.scope = (if provider.respond_to?(:scope) && provider.scope
+                                            provider.scope
+                                          end) || DEFAULT_SCOPE
```

Comment:
This is a place where you've mixed single line and multiline changes. I'm not keen on that. Single line was kind of fine (except for that IMHO nasty inline assignment to `...scope`) but if you really want to make this multiline, please do that for all of the `if`s. The same hanging right-hand column alignment comments (for that reformatted `scope` assignment) apply here as above, too.

For example, if we were going all-in on a tidy-up here coupled with the parentheses fix on `defined?`, then it might look like this; the `scope` change isn't 100% equivalent but I'm fairly sure it'll yield the same outcome in practice.

```ruby
if provider.respond_to?(:domain_hint) && provider.domain_hint
  options.authorize_params.domain_hint = provider.domain_hint
end

if defined?(request) && request.params['prompt']
  options.authorize_params.prompt = request.params['prompt']
end

if provider.respond_to?(:scope) && provider.scope
  options.authorize_params.scope = provider.scope
end

options.authorize_params.scope ||= DEFAULT_SCOPE
```

Reply:
Im down to make it all multi-line!